### PR TITLE
[BEAM-2727] Make beam builds reproducible

### PR DESCRIPTION
**Pull request by iemejia**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by iemejia** [comment]
R: @kennknowles 
CC: @jbonofre

File: pom.xml [hunk 1]
```diff
@@ -1760,6 +1767,19 @@
...
         </plugin>
+
+        <plugin>
+          <groupId>io.github.zlika</groupId>
+          <artifactId>reproducible-build-maven-plugin</artifactId>
+          <version>${reproducible-build-maven-plugin.version}</version>
+          <executions>
+            <execution>
+              <goals>
+                <goal>strip-jar</goal>
```
Comment (kennknowles) [inline pom.xml]: Can you describe the differences to the jar? I tried a little of this a while ago and was not sure about what was OK to remove.

Comment (iemejia) [inline pom.xml]: Well the thing is that java adds timestamps in some of the jar elements, e.g. the MANIFEST file and in some cases in the files that are updated with last modification date. The plugin covers these and other cases, you can take a look at the presentation he did (is really short) for some more detail. https://zlika.github.io/presentations/devoxx_fr_2016/reproducible-builds/slides_en.html#/_pom_properties

File: runners/apex/src/test/java/org/apache/beam/runners/apex/ApexYarnLauncherTest.java [hunk 2]
```diff
@@ -134,6 +134,7 @@ public void testCreateJar() throws Exception {
       Assert.assertTrue("manifest", Files.isRegularFile(zipfs.getPath(JarFile.MANIFEST_NAME)));
       Assert.assertTrue("file1", Files.isRegularFile(zipfs.getPath(file1)));
     }
-
+    // delete the directory to not let stuff around after the test execution
+    FileUtils.deleteDirectory(baseDir);
```
Comment (kennknowles) [inline runners/apex/src/test/java/org/apache/beam/runners/apex/ApexYarnLauncherTest.java]: Probably this should use `@Rule TemporaryFolder`

Comment (iemejia) [inline runners/apex/src/test/java/org/apache/beam/runners/apex/ApexYarnLauncherTest.java]: Good idea, done.

**Comment by iemejia** [comment]
I agree with you, I have some preference for more community maintained plugins however this one makes the job and I reported one issue that was fixed and released really quickly so I think we can give it a try.
Please don't resolve/close the JIRA issue if this works because I would like still to add the JDK/maven information to the template vote. (I wonder if proposing the releaser to use the docker developer image might solve the reproduciblity issue, as well to force us to keep the docker image updated).

**Comment by kennknowles** [comment]
LGTM. Be sure to mention this when the next release thread comes up, as I guess if anything is going to show up as a problem, it will be in the built artifacts then.

**Pull request closed without merging**